### PR TITLE
feat: Claim rewards

### PR DESCRIPTION
**Pull request by aballerr**
https://uniswaplabs.atlassian.net/browse/WEB-2318

### CODE REVIEW

**Comment by vm** [comment]
can you merge main real quick? think there's some conflicts

File: src/components/AidropModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+import Modal from '../Modal'
+
+const ModalWrap = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+
+const Body = styled.div`
+  padding-left: 20px;
```
Comment (vm) [inline src/components/AidropModal/index.tsx:20]: nit: padding: 20px

File: src/components/AidropModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  padding-top: 28px;
+  padding-bottom: 20px;
+`
+
+const ClaimButton = styled(ThemeButton)`
+  width: 100%;
+  background-color: ${({ theme }) => theme.accentAction};
+  margin-top: 40px;
+  border-radius: 12px;
+  color: white;
```
Comment (vm) [inline src/components/AidropModal/index.tsx:31]: think you want to use theme here?

File: src/components/AidropModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+`
+
+const Line = styled.div`
```
Comment (vm) [inline src/components/AidropModal/index.tsx:34]: can this just be a border?

File: src/state/application/reducer.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ export type PopupContent =
 
 export enum ApplicationModal {
   ADDRESS_CLAIM,
+  USDC_AIRDROP_CLAIM,
```
Comment (vm) [inline src/state/application/reducer.ts:18]: can you call this UNISWAP_NFT_AIRDROP_CLAIM

File: src/theme/components/text.tsx [hunk 5]
```diff
@@ -18,6 +18,15 @@ export const ThemedText = {
   BodyPrimary(props: TextProps) {
     return <TextWrapper fontWeight={400} fontSize={16} color="textPrimary" {...props} />
   },
+  BodyCaption(props: TextProps) {
+    return <TextWrapper fontWeight={400} fontSize={12} color="white" lineHeight="16px" {...props} />
+  },
+  BodyBodySmall(props: TextProps) {
```
Comment (vm) [inline src/theme/components/text.tsx:24]: isn't this just body small?

File: src/theme/components/text.tsx [hunk 6]
```diff
@@ -18,6 +18,15 @@ export const ThemedText = {
...
+    return <TextWrapper fontWeight={400} fontSize={14} lineHeight="20px" color="textPrimary" {...props} />
+  },
+  ButtonLabelMedium(props: TextProps) {
```
Comment (vm) [inline src/theme/components/text.tsx:27]: hm. idk about this name. should it just be label medium?

File: src/components/AidropModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+                </div>
+              </ThemedText.Link>
+            </EtherscanLink>
+
+            <CloseButton size={ButtonSize.medium} emphasis={ButtonEmphasis.medium} onClick={dismiss}>
+              Close
+            </CloseButton>
+          </ClaimContainer>
+        ) : (
+          <>
```
Comment (vm) [inline src/components/AidropModal/index.tsx:169]: you don't need this `<>`

**Review by vm — APPROVED** [review]
LGTM just confirm about the text color stuff and maybe remove the Line component in favor of a border

File: src/components/AidropModal/index.tsx (hunk 7 above)
Comment (aballerr) [reply src/components/AidropModal/index.tsx:169]: I get an error if I remove

File: src/components/AidropModal/index.tsx (hunk 3 above)
Comment (aballerr) [reply src/components/AidropModal/index.tsx:34]: maybe, but I personally find this simpler

File: .eslintrc.json [hunk 8]
```diff
@@ -30,7 +30,8 @@
     ".idea/",
     ".vscode/",
     "package-lock.json",
-    "yarn.lock"
+    "yarn.lock",
+    "src/*"
```
Comment (vm) [inline .eslintrc.json:34]: rm

Comment (vm) [reply .eslintrc.json:34]: why do you do this btw? surprised you don't like linting during dev

File: src/components/AidropModal/index.tsx (hunk 3 above)
Comment (vm) [reply src/components/AidropModal/index.tsx:31]: that's fine

**Pull request merged**